webCommander/centralHub/spotify/spotifyAPI.py

```diff
@@ -48,35 +52,59 @@ def get_spotify_artist(artist_name):
     Authenticate then search an artist based on an artist name. Searches for the name if id is not provided
     '''
 
-    if len(artist_name) != 22:
-        artist_id = search_spotify_artist(artist_name)
-
-    headers = spotify_authenticate()
+    artist_id = search_spotify_artist(artist_name)
 
     url = SPOTIFY_BASE_URL + SPOTIFY_ARTISTS + artist_id
-    response = requests.get(url, headers=headers)
+
+    response = requests.get(url, headers=spotify_authenticate())
     return response.text
 
 def search_spotify_artist(artist_name):
+    results = []
 
     url = SPOTIFY_BASE_URL + SPOTIFY_SEARCH
 
-    headers = spotify_authenticate()
-
     params = {
         "q": artist_name,
         "type": "artist"
     }
 
-    response = requests.get(url, headers=headers, params=params)
+    response = requests.get(url, headers=spotify_authenticate(), params=params)
     artists_results = json.loads(response.text)
 
+    #TODO Return a list of artists for a dropdown menu
     for artist in artists_results['artists']['items']:
-        if artist['name'] == artist_name:
-            return artist['id']
+        name = artist['name']
+        popularity = artist['popularity']
+        if len(artist['images']) > 0:
+            album_image = artist['images'][0]['url']
+        else:
+            album_image = "N/A"
+        results.append((name,album_image,popularity))
+
+    return results
+
+def get_artist_albums(artist_name):
+    album_dict = defaultdict(list)
+
+    artist_id = search_spotify_artist(artist_name)
+
+    url = SPOTIFY_BASE_URL + SPOTIFY_ARTISTS + artist_id + '/albums'
+
+    response = requests.get(url, headers=spotify_authenticate())
+    albums_results = json.loads(response.text)
+
+    #TODO
+    #not sure if this is the best data type for this purpose
```

Comment:
is this ok now or do we need a refactoring here? let me know if you have a question/doubt here ...

Reply:
This is OK. Will remove the tag on the next PR. I think its working very well.